kythe/extractors/openjdk11/extract.go

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2019 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"bufio"
+	"errors"
+	"flag"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"kythe.io/kythe/go/util/flagutil"
+
+	"github.com/bazelbuild/rules_go/go/tools/bazel"
+)
+
+const (
+	javaCommandVar      = "KYTHE_JAVA_COMMAND"
+	kytheRootVar        = "KYTHE_ROOT_DIRECTORY"
+	kytheVnameVar       = "KYTHE_VNAMES"
+	javaMakeVar         = "JAVA_CMD"
+	runfilesWrapperPath = "io_kythe/kythe/extractors/openjdk11/java-wrapper"
+	runfilesVnamesPath  = "io_kythe/kythe/extractors/openjdk11/vnames.json"
+)
+
+var (
+	wrapperPath  string
+	vnameRules   string
+	errorPattern = regexp.MustCompile("ERROR: extractor failure for module ([^:]*):")
+)
+
+func setupRunfiles() error {
+	if len(os.Getenv("RUNFILES_DIR")) > 0 || len(os.Getenv("RUNFILES_MANIFEST_FILE")) > 0 {
```

Comment:
```go
if os.Getenv("RUNFILES_DIR") != "" || os.Getenv("RUNFILES_MANIFEST_FILE") != "" {
```

Alternatively, if you care about the set/empty distinction you can use https://godoc.org/os#LookupEnv

Reply:
Yeah, I'm specifically not caring about the difference between unset and empty, because empty is always going to be an invalid value anyway.

Reply:
Ideally, the go bazel runfiles library would have the same semantics as the C++/Java ones (which would obviate all of this) but noooooo.